python/cudaq/utils/Tensor.py

```diff
@@ -0,0 +1,257 @@
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+import numpy as np
+from ..mlir._mlir_libs._quakeDialects.cudaq_runtime import utils
+
+
+class Tensor:
+    """
+    A multi-dimensional array class for CUDA-QX quantum computing libraries.
+
+    This class wraps CUDA-QX core tensor types and provides a NumPy-like interface
+    for quantum tensor operations.
+
+    Attributes:
+        _tensor: The underlying CUDA-Q core tensor object.
+        _dtype: The data type of the tensor elements.
+    """
+
+    def __init__(self, data=None, shape=None, dtype=None):
+        """
+        Initialize a Tensor object.
+
+        Args:
+            data (array-like, optional): Initial data for the tensor. Can be a list, tuple, 
+                                         numpy array, or CUDA-Q core tensor.
+            shape (tuple, optional): Shape of the tensor if creating an empty tensor.
+            dtype (numpy.dtype, optional): Data type of the tensor elements.
+
+        Raises:
+            RuntimeError: If initializing from a CUDA-Q core tensor without specifying dtype.
+            ValueError: If data is not a supported type or if dtype is unsupported.
+        """
+        self._tensor = None
+        self._dtype = None
+
+        if isinstance(data,
+                      (utils.Tensor_float, utils.Tensor_int32, utils.Tensor_int64,
+                       utils.Tensor_complex64, utils.Tensor_complex128, utils.Tensor_uint8)):
+            if dtype is None:
+                raise RuntimeError(
+                    "Tensor init from CUDA-QX Tensor must provide dtype.")
+            self._tensor = data
+            self._dtype = dtype
+            return
+
+        if data is not None:
+            if isinstance(data, (list, tuple, np.ndarray)):
+                data = np.array(data)
+                if dtype is None:
+                    dtype = data.dtype
+            else:
+                raise ValueError("Data must be a list, tuple, or numpy array")
+
+            self._create_tensor(data, dtype)
+        elif shape is not None:
+            if dtype is None:
+                dtype = np.float64
+            self._create_empty_tensor(shape, dtype)
+        else:
+            if dtype is None:
+                dtype = np.float64
+            self._create_empty_tensor([], dtype)
+
+    def _create_tensor(self, data, dtype):
+        """
+        Create a tensor from given data and dtype.
+
+        Args:
+            data (numpy.ndarray): The data to initialize the tensor with.
+            dtype (numpy.dtype): The data type of the tensor elements.
+
+        Raises:
+            ValueError: If the dtype is unsupported.
+        """
+        self._dtype = dtype
+        if dtype == np.float64:
```

Comment:
float32, int16 ?

Reply:
definitely ok to add more supported types. Just make sure to add them in py_tensor.cpp and in the bottom of xtensor_impl.cpp